src/elm/Search.elm

```diff
@@ -0,0 +1,478 @@
+
+
+offersSelectionSet : SelectionSet Offer Cambiatus.Object.Product
+offersSelectionSet =
+    SelectionSet.map4 Offer
+        Cambiatus.Object.Product.id
+        Cambiatus.Object.Product.title
+        Cambiatus.Object.Product.price
+        Cambiatus.Object.Product.image
+
+
+
+-- UPDATE
+
+
+type Msg
+    = CancelClicked
+    | InputFocused
+    | GotRecentSearches String
+    | RecentQueryClicked String
+    | SearchResultsLoaded (Result (Graphql.Http.Error SearchResult) SearchResult)
+    | QuerySubmitted
+    | TabActivated ActiveTab
+    | CurrentQueryChanged String
+    | FoundItemClicked Route
+
+
+closeSearch : Shared -> Model -> ( Model, Cmd Msg )
+closeSearch shared model =
+    update shared model CancelClicked
+
+
+update : Shared -> Model -> Msg -> ( Model, Cmd Msg )
+update shared model msg =
+    case msg of
+        CurrentQueryChanged q ->
+            ( { model | currentQuery = q }, Cmd.none )
+
+        FoundItemClicked route ->
+            ( { model
+                | state = Inactive
+                , currentQuery = ""
+              }
+            , Route.replaceUrl shared.navKey route
+            )
+
+        RecentQueryClicked q ->
+            update shared
+                { model
+                    | state = Loading
+                    , currentQuery = q
+                }
+                QuerySubmitted
+
+        TabActivated activeTab ->
+            case model.state of
+                ResultsShowed r _ ->
+                    ( { model
+                        | state = ResultsShowed r activeTab
+                      }
+                    , Cmd.none
+                    )
+
+                _ ->
+                    ( model, Cmd.none )
+
+        SearchResultsLoaded res ->
+            case res of
+                Ok searchResult ->
+                    ( { model
+                        | state = ResultsShowed (Just searchResult) ResultsOverview
+                      }
+                    , Cmd.none
+                    )
+
+                Err _ ->
+                    ( model, Cmd.none )
+
+        GotRecentSearches queries ->
+            case Decode.decodeString (list string) queries of
+                Ok queryList ->
+                    ( { model | recentQueries = queryList }, Cmd.none )
+
+                Err _ ->
+                    ( model, Cmd.none )
+
+        CancelClicked ->
+            ( { model
+                | state = Inactive
+                , currentQuery = ""
+              }
+            , Cmd.none
+            )
+
+        InputFocused ->
+            ( { model
+                | state = RecentSearchesShowed
+              }
+            , Cmd.none
+            )
+
+        QuerySubmitted ->
+            let
+                newRecentSearches : List String
+                newRecentSearches =
+                    (model.currentQuery :: model.recentQueries)
+                        |> List.unique
+                        |> List.take 3
```

Comment:
only 3?

Reply:
Yep, there's a note on Figma:
![image](https://user-images.githubusercontent.com/140053/107746549-0c278200-6d27-11eb-92d4-277875447b64.png)

Anyway, we always can change the number.